libphal/phal_dump.C

```diff
@@ -0,0 +1,418 @@
+		int err = errno;
+		log(level::ERROR,
+		    "Failed to write to dump file err=%d error message=%s", err,
+		    oe.what());
+		throw dumpError_t(exception::FILE_OPERATION_FAILED);
+	}
+	outfile.close();
+}
+
+void collectSBEDump(uint32_t id, uint32_t failingUnit,
+		    const std::filesystem::path& dumpPath)
+{
+	log(level::INFO,
+	    "Collecting SBE dump: path=%s, id=%d, chip position=%d",
+	    dumpPath.string().c_str(), id, failingUnit);
+
+	std::stringstream ss;
+	ss << std::setw(8) << std::setfill('0') << id;
+
+	std::string baseFilename =
+	    ss.str() + ".0_" + std::to_string(failingUnit) + "_SbeData_p10_";
+
+	// Execute pre-collection and get proc corresponding to failing unit
+	struct pdbg_target* proc = preCollection(failingUnit);
+
+	// Get pib for the proc
+	struct pdbg_target* pib = getPibTarget(proc);
+
+	// Set SBE state to SBE_STATE_DEBUG_MODE
+	sbe_set_state(pib, SBE_STATE_DEBUG_MODE);
+
+	fapi2::ReturnCode fapiRc;
+
+	try {
+		// Collect SBE local register dump
+		std::vector<SBESCOMRegValue_t> sbeScomRegValue;
+		fapiRc = p10_sbe_localreg_dump(proc, true, sbeScomRegValue);
+		if (fapiRc != fapi2::FAPI2_RC_SUCCESS) {
+			log(level::ERROR,
+			    "Failed in p10_sbe_localreg_dump for proc=%s, "
+			    "rc=0x%08X",
+			    pdbg_target_path(proc), fapiRc);
+		} else {
+			std::vector<DumpSBERegVal> dumpRegs;
+			for (auto& reg : sbeScomRegValue) {
+				dumpRegs.emplace_back(reg.reg.number,
+						      reg.reg.name, reg.value);
+			}
+			std::string dumpFilename =
+			    baseFilename + "p10_sbe_localreg_dump";
+			std::filesystem::path basePath =
+			    dumpPath / dumpFilename;
+
+			try {
+				writeDumpFile(
+				    reinterpret_cast<char*>(&dumpRegs[0]),
+				    sizeof(DumpSBERegVal) * dumpRegs.size(),
+				    basePath);
+			} catch (const dumpError_t& e) {
+				log(level::ERROR,
+				    "Error in writing p10_sbe_localreg_dump "
+				    "file "
+				    "errorMsg=%s",
+				    e.what());
+				throw;
+			}
+		}
+
+		// Dump contents of various PIB Masters and Slaves internal
+		// registers
+		std::vector<sRegV> pibmsRegSet;
+		for (auto& reg : pibms_regs_2dump) {
+			sRegV regv;
+			regv.reg = reg;
+			pibmsRegSet.emplace_back(regv);
+		}
+
+		fapiRc = p10_pibms_reg_dump(proc, pibmsRegSet);
+		if (fapiRc != fapi2::FAPI2_RC_SUCCESS) {
+			log(level::ERROR,
+			    "Failed in p10_pibms_reg_dump for proc=%s, "
+			    "rc=0x%08X",
+			    pdbg_target_path(proc), fapiRc);
+		} else {
+			std::vector<DumpPIBMSRegVal> dumpRegs;
+			for (sRegV& regs : pibmsRegSet) {
+				dumpRegs.emplace_back(
+				    regs.reg.addr, regs.reg.name, regs.reg.attr,
+				    regs.value);
+			}
+			std::string dumpFilename =
+			    baseFilename + "p10_pibms_reg_dump";
+			std::filesystem::path basePath =
+			    dumpPath / dumpFilename;
+			try {
+				writeDumpFile(
+				    reinterpret_cast<char*>(&dumpRegs[0]),
+				    sizeof(DumpPIBMSRegVal) * dumpRegs.size(),
+				    basePath);
+			} catch (const dumpError_t& e) {
+				log(level::ERROR,
```

Comment:
Leving SBE state to debug mode , this can lead confusion , better change state to failled?
Note Applkicable for all failing cases .

Reply:
This will not, the outer catch block is setting the state to failed